provider/aci.go

```diff
@@ -1635,6 +1635,37 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 func (p *ACIProvider) getVolumes(pod *v1.Pod) ([]aci.Volume, error) {
 	volumes := make([]aci.Volume, 0, len(pod.Spec.Volumes))
 	for _, v := range pod.Spec.Volumes {
+		// Handle the case for Azure File CSI driver
+		if v.CSI != nil {
+			pv, err := p.resourceManager.GetPersistentVolume(v.Name)
+			if err != nil {
+				return volumes, err
+			}
+			csiSource := pv.Spec.CSI
+
+			if csiSource.NodeStageSecretRef != nil && csiSource.NodeStageSecretRef.Name != "" {
+				secret, err := p.resourceManager.GetSecret(csiSource.NodeStageSecretRef.Name, csiSource.NodeStageSecretRef.Namespace)
+				if err != nil {
+					return volumes, err
+				}
+
+				if secret == nil {
+					return nil, fmt.Errorf("Getting secret for AzureFile CSI driver returned an empty secret")
+				}
+				volume := aci.Volume{
+					Name: v.Name,
+					AzureFile: &aci.AzureFileVolume{
+						ReadOnly:           csiSource.ReadOnly,
+						StorageAccountName: string(secret.Data["azurestorageaccountname"]),
+						StorageAccountKey:  string(secret.Data["azurestorageaccountkey"]),
+					},
+				}
+				volumes = append(volumes, volume)
+			}
+
+			continue
+		}
```

Comment:
I know the current code didn't do it but I think we should extract this into its own function.  We can refactor the rest piece by piece